GoogleMapsComponents/wwwroot/js/objectManager.js

```diff
@@ -463,6 +500,20 @@
                         delete mapObjects[elementToRemove];
                     }
                 }
+
+                if(controlParents.hasOwnProperty(mapGuid)){
+                    for (var key in controlParents[mapGuid]) {
+                        if (controlParents[mapGuid].hasOwnProperty(key)) {
```

Comment:
Code duplication. Please try some other method which could be reused

Reply:
IMO this must be refactored. Other parts are very minor

Reply:
Will do.

Reply:
Refactored in new commit